crates/uv/src/commands/tool/list.rs

```diff
@@ -27,8 +27,19 @@ pub(crate) async fn list(preview: PreviewMode, printer: Printer) -> Result<ExitS
     }
 
     // TODO(zanieb): Track and display additional metadata, like entry points
-    for (name, _tool) in tools {
+    for (name, tool) in tools {
+        // Output tool name
         writeln!(printer.stdout(), "{name}")?;
+
+        // Output tool entrypoints
+        for entrypoint in tool
+            .entrypoints()
+            .iter()
+            .map(|entry| entry.name.clone())
+            .collect::<Vec<String>>()
```

Comment:
Why collect here instead of just doing `for entrypoint in tool.entrypoints() { writeln(...) }`?

Reply:
indeed i have overcomplicated this a lot, the new [commit](https://github.com/astral-sh/uv/pull/4661/commits/94e79061197ed3cdfb651c962e2827c661c834d7) should be better